plugins/extractors/utils/sql_dbs.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"database/sql"
+
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/utils"
+)
+
+// Config holds the connection URL for the extractor
+type Config struct {
+	ConnectionURL string `mapstructure:"connection_url" validate:"required"`
+}
+
+// Extractor manages the extraction of data from MySQL
+type BaseExtractor struct {
```

Comment:
This is more like an `BaseSQLExtractor` more than `BaseExtractor`.
An idea is to have `BaseSQLExtractor` it embedded `BaseExtractor`, but I'm still not sure if it is a good idea.
wdyt?

Reply:
I am not sure if it needs these many Different structs, cause there are not many fields in Extractor struct.

Reply:
I am thinking should it be a struct or can we actually just have pure util functions?

Reply:
Hi @GrayFlash, I don't think that we need that much generalization/abstraction by pulling this into a struct called `BaseExtractor`. Our `Plugin` interface and `Extractor` interface are abstract enough to define the contract.

It is true those functions are repetitive (the one in BaseExtractor). We could just have some pure and generic function to do that functions. 

For example, rather than having a function called `BuildExcludedDBs`, maybe we can generalize it to `BuildExclusions` / `BuildStringExclusion`.

We could improve the util package naming by not naming it with generic name like `util` but more into with the name that indicate the purpose of functions inside it.